Lib/test/test_imaplib.py

```diff
@@ -879,6 +879,14 @@ def test_with_statement_logout(self):
                 self.assertIsNone(server.logged)
             self.assertIsNone(server.logged)
 
+    @reap_threads
+    def test_noop_with_debug(self):
```

Comment:
For this test to work, there needs to be some untagged response from the server. 

A simpler test case IMO, which can reproduce the issue is to have a simple untagged_response_dict passed directly to `_dump_ur` and make sure that it is correctly able to handle bytes data.

Reply:
Yeah, I concur: test directly _dump_ur(). But since it's a private method, please decorate the test with @support.cpython_only since _dump_ur() is an implementation detail.